File: tools/integrity/tests/fixtures/good_contracts_b/common/common-web/src/widget.ts
export class Widget {
    name: string;
    count: number;

    constructor(name: string) {
        this.name = name;
        this.count = 0;
    }

    increment(): void {
        this.count += 1;
    }
}

export function makeWidget(name: string): Widget {
    return new Widget(name);
}

File: tools/integrity/tests/fixtures/bad_stub_label/common/common-py/gpusims_common/widget.py
"""Widget module.

In Phase 1, this is a stub: callers should expect partial functionality.
"""


def make_widget(count: int) -> dict:
    out = {}
    out["count"] = count
    out["positions"] = [0.0] * (3 * count)
    out["velocities"] = [0.0] * (3 * count)
    out["radii"] = [1.0] * count
    out["ids"] = list(range(count))
    out["timestamps"] = [0.0] * count
    return out


def consume_widget(w: dict) -> int:
    return w["count"] * 2 + len(w["positions"])
